src/project/internal/projectactionscontroller.cpp

```diff
@@ -371,13 +406,103 @@ void ProjectActionsController::downloadAndOpenCloudProject(int scoreId)
             return;
         }
 
-        doOpenCloudProject(localPath, info);
+        doOpenCloudProject(localPath, info, isOwner);
     });
 
     m_projectBeingDownloadedChanged.notify();
     isDownloadingFinished = false;
 }
 
+Ret ProjectActionsController::openMuseScoreUrl(const QUrl& url)
+{
+    if (url.host() == OPEN_SCORE_URL_HOSTNAME) {
+        return openScoreFromMuseScoreCom(url);
+    }
+
+    return make_ret(Err::UnsupportedUrl);
+}
+
+Ret ProjectActionsController::openScoreFromMuseScoreCom(const QUrl& url)
+{
+    //! NOTE See explanation in `openProject(const io::path_t& _path, const QString& displayNameOverride)`
+    if (m_isProjectProcessing || m_isProjectDownloading) {
+        // TODO: instead of ignoring the open request, queue it?
+        return make_ret(Ret::Code::InternalError);
+    }
+    m_isProjectProcessing = true;
+
+    DEFER {
+        m_isProjectProcessing = false;
+    };
+
+    // Retrieve score id from URL
+    bool ok = false;
+    int scoreId = url.fileName().toInt(&ok);
+    if (!ok || scoreId <= 0) {
+        return make_ret(Err::MalformedOpenScoreUrl);
+    }
+
+    // Ensure logged in
+    Ret ret = museScoreComService()->authorization()->ensureAuthorization(
+        trc("project/save", "Login or create a free account on musescore.com to open this score."));
+    if (!ret) {
+        return ret;
+    }
+
+    // Check if user is owner
+    bool isOwner = true;
+    RetVal<cloud::ScoreInfo> scoreInfo = museScoreComService()->downloadScoreInfo(scoreId);
+    if (!scoreInfo.ret) {
+        std::any status = scoreInfo.ret.data("status");
+        if (status.has_value() && std::any_cast<int>(status) == 403) {
```

Comment:
It is better to return an error code and check for it instead of checking the status